engine/src/flutter/lib/web_ui/lib/src/engine/semantics/semantics.dart

```diff
@@ -810,6 +818,15 @@ abstract class SemanticRole {
     removeAttribute('aria-controls');
   }
 
+  void _updateLocale() {
+    final String locale = semanticsObject.locale?.toString() ?? '';
+    if (locale.isEmpty) {
+      removeAttribute('lang');
+      return;
+    }
+    setAttribute('lang', locale);
+  }
```

Comment:
Can we do something like this to avoid setting the `lang` attribute everywhere:
```suggestion
  void _updateLocale() {
    final String locale = semanticsObject.locale?.toString() ?? '';
    final isSameAsParent = semanticsObject.locale == semanticsObject.parent?.locale;
    if (locale.isEmpty || isSameAsParent) {
      removeAttribute('lang');
      return;
    }
    setAttribute('lang', locale);
  }
```